CONTRIBUTING.md

```diff
@@ -129,9 +150,43 @@ Now cd to ~/repos/NVIDIA/spark-rapids and build per
 $ ./build/buildall
 ```
 
-Prior to switching to another spark-rapids-jni or submiting a PR to NVIDIA/spark-rapids-jni,
-we should undo the cudf submodule modification.
+Since we rely on local Maven cache we need to pay extra attention to make sure that
+the final rapids-4-spark artifact included locally built dependencies as opposed to
+CI-built snapshot dependencies from the remote Maven repo. This may happen if Maven
+is invoked with `--offline` or `--no-snapshot-updates` option due to IDE-Maven
+interactions in the background. To confirm that the artifact is correct we can either enable INFO logging in Spark or directly inspect the resulting jar for build info:
```

Comment:
Nit: This line isn't wrapped which sticks out (literally) when viewing it outside of the github render (e.g.: from a terminal):
```suggestion
interactions in the background. To confirm that the artifact is correct we can either
enable INFO logging in Spark or directly inspect the resulting jar for build info:
```